### DISCUSSION TITLE
Applying many distinct rotations to EBSD data efficiently?

### DISCUSSION THREAD

**Discussion by gsparks3** [mtex-toolbox/mtex#1169] [Ask Anything] [answered]
I have a large array of EBSD data, and an equally large array of rotations. For each point of EBSD data, I wish to apply the corresponding rotation from said array. Currently I do this using a for-loop, indexing out each individual point of EBSD data and individual rotation and applying them, but the massive number of data-indexing calls makes this process very slow. Is there any more efficient way to do this?

I know for applying a _single_ rotation to many EBSD data points, the standard MTEX rotate() function can do this very efficiently, but I need to apply a unique rotation to each EBSD data point.

**Comment by kilir** [comment, marked as answer]
Doesn't `o_new= list_of_rots .* ebsd(xy).orientations` work? You can set the orientations afterwards e.g `ebsd(xy).orientations=o_new`.
Hope this helps.
Cheers,
Rüdiger

**Reply by gsparks3** [reply]
That does work! Thank you so much! It's always the little things ...
